src/agent/reqwest-retry/src/lib.rs

```diff
@@ -192,23 +192,20 @@ mod test {
 
     #[tokio::test]
     async fn retry_should_fail() -> Result<()> {
-        let invalid_url = "http://localhost:81/test.txt";
+        let invalid_url = "http://127.0.0.1:81/test.txt";
         let resp = reqwest::Client::new()
             .get(invalid_url)
             .send_retry(
                 Duration::from_secs(1),
-                Duration::from_secs(3),
-                3i32,
+                Duration::from_secs(300),
+                2i32,
                 to_backoff_response,
             )
             .await;
 
         if let Err(err) = &resp {
             let as_text = format!("{}", err);
-            assert!(
-                as_text.contains("Maximum number of attempts reached for this request")
-                    || as_text.contains("os error 10061")
-            );
+            assert!(as_text.contains(MAX_RETRY_ERROR_MESSAGE));
```

Comment:
I'm assuming this reversion was you root-causing the test failure and removing a recently-added band-aid.

Reply:
10061 was the error I was seeing locally which is WSAECONNREFUSED. I guessed (perhaps incorrectly) that retry wouldn't happen after this error.

Reply:
That's legit, just making sure that the churn here is intentional!

Reply:
The underlying WSAECONNREFUSED which @lzybkr was hitting is handled by the retry code.  

However, we were hitting "maximum duration for testing" instead of "tried too many times".  This PR adjusts the test such that we always hit the former.